contracts/deployables/freeze-voting/FreezeVotingStandaloneV1.sol

```diff
@@ -0,0 +1,519 @@
+        uint256 lightAccountIndex_
+    ) public virtual override {
+        // Check if already frozen
+        if (isFrozen()) revert AlreadyFrozen();
+
+        // Resolve voter address (handles light account)
+        address voter = potentialLightAccountResolvedOwner(
+            msg.sender,
+            lightAccountIndex_
+        );
+
+        // Check if proposal is expired and create new one if needed
+        FreezeVotingBaseStorage storage $base = _getFreezeVotingBaseStorage();
+        if (
+            block.timestamp >
+            $base.freezeProposalCreated + $base.freezeProposalPeriod
+        ) {
+            _initializeFreezeVote();
+            emit FreezeProposalCreated(uint48(block.timestamp), voter);
+        }
+
+        // Calculate total weight from all configs
+        uint256 totalWeight = _getVotes(voter, votingConfigsToUse_);
+
+        // Record the vote and handle freeze activation if threshold is reached
+        _recordFreezeVote(voter, totalWeight);
+    }
+
+    /**
+     * @inheritdoc IFreezeVotingStandaloneV1
+     */
+    function castUnfreezeVote(
+        IVotingTypes.VotingConfigVoteData[] calldata votingConfigsToUse_,
+        uint256 lightAccountIndex_
+    ) public virtual override {
+        FreezeVotingStandaloneStorage
+            storage $ = _getFreezeVotingStandaloneStorage();
+
+        // Check if frozen
+        if (!isFrozen()) revert NotFrozen();
+
+        // Resolve voter address
+        address voter = potentialLightAccountResolvedOwner(
+            msg.sender,
+            lightAccountIndex_
+        );
+
+        // Check if proposal is expired and create new one if needed
+        if (
+            $.unfreezeProposalCreated != 0 &&
+            block.timestamp >
+            $.unfreezeProposalCreated + $.unfreezeProposalPeriod
+        ) {
+            // Reset expired proposal
+            // Use previous timestamp to ensure ERC5805 getPastVotes works
+            $.unfreezeProposalCreated = uint48(block.timestamp - 1);
+            $.unfreezeProposalVoteCount = 0;
+        }
+
+        // Initialize new proposal if needed
+        if ($.unfreezeProposalCreated == 0) {
+            // Use previous timestamp to ensure ERC5805 getPastVotes works
+            $.unfreezeProposalCreated = uint48(block.timestamp - 1);
+            emit UnfreezeProposalCreated(uint48(block.timestamp - 1), voter);
+        }
+
+        // Calculate weight from configs
+        uint256 totalWeight = _getUnfreezeVotes(
+            voter,
+            $.unfreezeProposalCreated,
+            votingConfigsToUse_
+        );
+
+        // Check for zero weight (happens when all configs were already used)
+        if (totalWeight == 0) revert NoVotes();
+
+        $.unfreezeProposalVoteCount += totalWeight;
+
+        emit UnfreezeVoteCast(voter, totalWeight);
+    }
+
+    /**
+     * @inheritdoc IFreezeVotingStandaloneV1
+     */
+    function unfreeze() public virtual override(IFreezeVotingStandaloneV1) {
```

Comment:
I don't think we need to declare the `(IFreezeVotingStandaloneV1)` here

Reply:
good point, removed